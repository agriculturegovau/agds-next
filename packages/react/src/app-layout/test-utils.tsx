import { Fragment } from 'react';
import { Text } from '../text';
import { NotificationBadge } from '../notification-badge';
import { VisuallyHidden } from '../a11y';
import {
	ChartLineIcon,
	EmailIcon,
	ExitIcon,
	HelpIcon,
	HomeIcon,
	SuccessIcon,
	SettingsIcon,
	FactoryIcon,
	AvatarIcon,
	ChevronsLeftIcon,
	MinusIcon,
} from '../icon';
import {
	DropdownMenuDivider,
	DropdownMenuItem,
	DropdownMenuItemLink,
	DropdownMenuPanel,
	DropdownMenuGroup,
	DropdownMenuGroupLink,
	DropdownMenuItemRadio,
} from '../dropdown-menu';

export const navigationItems = (businessName: string) => [
	{
		options: { disableGroupPadding: true },
		items: [
			{
				label: 'Back to my account',
				icon: ChevronsLeftIcon,
				href: '/account',
			},
		],
	},
	{
		options: { disableGroupPadding: true },
		items: [
			{
				label: (
					<Fragment>
						<Text fontWeight="bold" fontSize="xs">
							{businessName}
						</Text>
						<Text color="muted" fontSize="xs">
							ABN: 00 000 000 000
						</Text>
					</Fragment>
				),
			},
		],
	},
	[
		{
			label: 'Dashboard',
			icon: HomeIcon,
			href: '/',
		},
		{
			label: 'Establishments',
			icon: FactoryIcon,
			href: '/establishments',
			items: [
				{
					label: 'Canberra',
					icon: MinusIcon,
					href: '/establishments/canberra',
				},
				{
					label: 'Sydney',
					icon: MinusIcon,
					href: '/establishments/sydney',
				},
			],
		},
		{
			label: 'Data and Insights',
			icon: ChartLineIcon,
			href: '/intelligence',
		},
		{
			label: 'Compliance',
			icon: SuccessIcon,
			href: '/compliance',
			items: [
				{
					label: 'Audit',
					icon: MinusIcon,
					href: '/compliance/audit',
				},
				{
					label: 'Certificates',
					icon: MinusIcon,
					href: '/compliance/certificates',
				},
			],
		},
	],

	[
		{
			label: 'Help',
			icon: HelpIcon,
			href: '/help',
		},
	],
	[
		{
			label: 'Sign out',
			onClick: console.log,
			icon: ExitIcon,
		},
	],
];

export const exampleData = {
	userNames: {
		short: 'Ab Cd',
		regular: 'Toto Wolff',
		medium: 'Benjamin Harrington',
		long: 'Benjamin Alexander Harrington-Smythe',
	},
	businessNames: {
		short: ['Lorem', 'Ipsum', 'Company'],
		regular: ['Antfix', 'Produce Fresh', 'Organic Co'],
		medium: [
			'Malesuada lacus a tortor blandit fermentum phasellus',
			'Iaculis tortor duis ante nec risus elementum id ui',
			'Purus tortor lacus malesuada phasellus ipsum ex duis libero ante id',
		],
		long: [
			'Duis quis iaculis libero nec consectetur luctus ex in malesuada lacus a tortor blandit fermentum phasellus iaculis ipsum nec purus aliquet id elementum risus duis ante ui',
			'Duis ut a libero duis nisl ex lacus in luctus viverra iaculis iaculis tortor duis malesuada blandit purus diam ipsum ante nec risus elementum id ui nec quis aliquet phasellus fermentum',
			'Blandit iaculis iaculis quis ante diam viverra elementum ui risus nec luctus purus tortor lacus malesuada phasellus ipsum ex duis libero ante id Duis fermentum nisl aliquet duis in a nec ut consectetur',
		],
	},
};

export function ExampleAccountDropdown({
	businesses = exampleData.businessNames.regular,
	onBusinessChange,
	selectedBusinessName,
}: {
	businesses?: string[];
	onBusinessChange: (value: string) => void;
	selectedBusinessName: string;
}) {
	return (
		<DropdownMenuPanel palette="light">
			<DropdownMenuGroup label="Businesses">
				{businesses.map((businessName) => (
					<DropdownMenuItemRadio
						key={businessName}
						checked={businessName === selectedBusinessName}
						secondaryText="ABN 00 000 000 000"
						onClick={() => onBusinessChange?.(businessName)}
					>
						{businessName}
					</DropdownMenuItemRadio>
				))}
				<DropdownMenuGroupLink href="#">View all</DropdownMenuGroupLink>
			</DropdownMenuGroup>
			<DropdownMenuDivider />
			<DropdownMenuGroup label="My account">
				<DropdownMenuItemLink href="/profile" icon={AvatarIcon}>
					Profile
				</DropdownMenuItemLink>
				<DropdownMenuItemLink
					href="/messages"
					icon={EmailIcon}
					endElement={
						<span>
							<NotificationBadge tone="action" value={6} max={99} aria-hidden />
							<VisuallyHidden>, 6 unread</VisuallyHidden>
						</span>
					}
				>
					Messages
				</DropdownMenuItemLink>
				<DropdownMenuItemLink href="/account-settings" icon={SettingsIcon}>
					Account settings
				</DropdownMenuItemLink>
			</DropdownMenuGroup>
			<DropdownMenuDivider />
			<DropdownMenuItem onClick={() => console.log('sign out')} icon={ExitIcon}>
				Sign out
			</DropdownMenuItem>
		</DropdownMenuPanel>
	);
}
