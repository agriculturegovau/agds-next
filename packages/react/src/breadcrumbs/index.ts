'use client';
export * from './Breadcrumbs';
export * from './BreadcrumbsContainer';
export * from './BreadcrumbsDivider';
export * from './BreadcrumbsItem';
export * from './BreadcrumbsToggle';
