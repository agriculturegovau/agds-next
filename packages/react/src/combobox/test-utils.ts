export type Option = { label: string; value: string };

export const COUNTRY_OPTIONS = [
	'Afghanistan is a really long country that spans multiple lines cmon this can be way longer than that Afghanistan is a really long country that spans multiple lines cmon this can be way longer than that',
	'Albania',
	'Algeria',
	'Andorra',
	'Angola',
	'Antigua and Barbuda',
	'Argentina',
	'Armenia',
	'Australia',
	'Austria',
	'Azerbaijan',
	'Bahamas',
	'Bahrain',
	'Bangladesh',
	'Barbados',
	'Belarus',
	'Belgium',
	'Belize',
	'Benin',
	'Bhutan',
	'Bolivia',
	'Bosnia and Herzegovina',
	'Botswana',
	'Brazil',
	'Brunei',
	'Bulgaria',
	'Burkina Faso',
	'Burundi',
	"CÃ´te d'Ivoire",
	'Cabo Verde',
	'Cambodia',
	'Cameroon',
	'Canada',
	'Central African Republic',
	'Chad',
	'Chile',
	'China',
	'Colombia',
	'Comoros',
	'Congo (Congo-Brazzaville)',
	'Costa Rica',
	'Croatia',
	'Cuba',
	'Cyprus',
	'Czechia (Czech Republic)',
	'Democratic Republic of the Congo',
	'Denmark',
	'Djibouti',
	'Dominica',
	'Dominican Republic',
	'Ecuador',
	'Egypt',
	'El Salvador',
	'Equatorial Guinea',
	'Eritrea',
	'Estonia',
	'Eswatini (fmr. "Swaziland")',
	'Ethiopia',
	'Fiji',
	'Finland',
	'France',
	'Gabon',
	'Gambia',
	'Georgia',
	'Germany',
	'Ghana',
	'Greece',
	'Grenada',
	'Guatemala',
	'Guinea',
	'Guinea-Bissau',
	'Guyana',
	'Haiti',
	'Holy See',
	'Honduras',
	'Hungary',
	'Iceland',
	'India',
	'Indonesia',
	'Iran',
	'Iraq',
	'Ireland',
	'Israel',
	'Italy',
	'Jamaica',
	'Japan',
	'Jordan',
	'Kazakhstan',
	'Kenya',
	'Kiribati',
	'Kuwait',
	'Kyrgyzstan',
	'Laos',
	'Latvia',
	'Lebanon',
	'Lesotho',
	'Liberia',
	'Libya',
	'Liechtenstein',
	'Lithuania',
	'Luxembourg',
	'Madagascar',
	'Malawi',
	'Malaysia',
	'Maldives',
	'Mali',
	'Malta',
	'Marshall Islands',
	'Mauritania',
	'Mauritius',
	'Mexico',
	'Micronesia',
	'Moldova',
	'Monaco',
	'Mongolia',
	'Montenegro',
	'Morocco',
	'Mozambique',
	'Myanmar (formerly Burma)',
	'Namibia',
	'Nauru',
	'Nepal',
	'Netherlands',
	'New Zealand',
	'Nicaragua',
	'Niger',
	'Nigeria',
	'North Korea',
	'North Macedonia',
	'Norway',
	'Oman',
	'Pakistan',
	'Palau',
	'Palestine State',
	'Panama',
	'Papua New Guinea',
	'Paraguay',
	'Peru',
	'Philippines',
	'Poland',
	'Portugal',
	'Qatar',
	'Romania',
	'Russia',
	'Rwanda',
	'Saint Kitts and Nevis',
	'Saint Lucia',
	'Saint Vincent and the Grenadines',
	'Samoa',
	'San Marino',
	'Sao Tome and Principe',
	'Saudi Arabia',
	'Senegal',
	'Serbia',
	'Seychelles',
	'Sierra Leone',
	'Singapore',
	'Slovakia',
	'Slovenia',
	'Solomon Islands',
	'Somalia',
	'South Africa',
	'South Korea',
	'South Sudan',
	'Spain',
	'Sri Lanka',
	'Sudan',
	'Suriname',
	'Sweden',
	'Switzerland',
	'Syria',
	'Tajikistan',
	'Tanzania',
	'Thailand',
	'Timor-Leste',
	'Togo',
	'Tonga',
	'Trinidad and Tobago',
	'Tunisia',
	'Turkey',
	'Turkmenistan',
	'Tuvalu',
	'Uganda',
	'Ukraine',
	'United Arab Emirates',
	'United Kingdom',
	'United States of America',
	'Uruguay',
	'Uzbekistan',
	'Vanuatu',
	'Venezuela',
	'Vietnam',
	'Yemen',
	'Zambia',
	'Zimbabwe',
].map((country) => ({ label: country, value: country }));

export const STATE_OPTIONS = [
	{ label: 'Australian Capital Territory', value: 'act' },
	{ label: 'New South Wales', value: 'nsw' },
	{ label: 'Northern Territory', value: 'nt' },
	{ label: 'Queensland', value: 'qld' },
	{ label: 'South Australia', value: 'sa' },
	{ label: 'Tasmania', value: 'tas' },
	{ label: 'Victoria', value: 'vic' },
	{ label: 'Western Australia', value: 'wa' },
];

export const NAME_OPTIONS = [
	{
		fullName: 'John Doe',
		jobTitle: 'Product Manager',
		status: 'Draft',
		unreadMessageCount: 0,
	},
	{
		fullName: 'Jane Smith',
		jobTitle: 'Software Engineer',
		status: 'Approved',
		unreadMessageCount: 4,
	},
	{
		fullName: 'Robert Johnson',
		jobTitle: 'UX Designer',
		status: 'Deactivated',
		unreadMessageCount: 5,
	},
	{
		fullName: 'Emily Davis',
		jobTitle: 'Product Manager',
		status: 'Approved',
		unreadMessageCount: 6,
	},
	{
		fullName: 'Michael Williams',
		jobTitle: 'Software Engineer',
		status: 'Approved',
		unreadMessageCount: 0,
	},
	{
		fullName: 'Sarah Wilson',
		jobTitle: 'UX Designer',
		status: 'Approved',
		unreadMessageCount: 0,
	},
	{
		fullName: 'David Brown',
		jobTitle: 'Product Manager',
		status: 'Deactivated',
		unreadMessageCount: 0,
	},
	{
		fullName: 'Olivia Taylor',
		jobTitle: 'Software Engineer',
		status: 'Approved',
		unreadMessageCount: 6,
	},
	{
		fullName: 'James Moore',
		jobTitle: 'UX Designer',
		status: 'Approved',
		unreadMessageCount: 0,
	},
	{
		fullName: 'Sophia Clark',
		jobTitle: 'Product Manager',
		status: 'Approved',
		unreadMessageCount: 0,
	},
].map((option) => ({
	...option,
	label: option.fullName,
	value: option.fullName,
}));
