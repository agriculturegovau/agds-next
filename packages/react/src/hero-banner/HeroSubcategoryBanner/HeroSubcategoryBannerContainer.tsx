import { PropsWithChildren } from 'react';
import { Box } from '../../box';
import { HeroBannerBackground } from '../utils';

export type HeroSubcategoryBannerContainerProps = PropsWithChildren<{
	background: HeroBannerBackground;
}>;

export const HeroSubcategoryBannerContainer = ({
	children,
	background,
}: HeroSubcategoryBannerContainerProps) => {
	return (
		<Box as="section" background={background}>
			{children}
		</Box>
	);
};
