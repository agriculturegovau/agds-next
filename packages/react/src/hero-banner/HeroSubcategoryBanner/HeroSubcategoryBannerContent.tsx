import { ReactNode } from 'react';
import { Stack } from '../../stack';
import { Content } from '../../content';

export type HeroSubcategoryBannerContentProps = { children: ReactNode };

export const HeroSubcategoryBannerContent = ({
	children,
}: HeroSubcategoryBannerContentProps) => {
	return (
		<Content>
			<Stack
				gap={1.5}
				paddingBottom={{ xs: 3, md: 3 }}
				paddingTop={{ xs: 2, md: 3 }}
			>
				{children}
			</Stack>
		</Content>
	);
};
