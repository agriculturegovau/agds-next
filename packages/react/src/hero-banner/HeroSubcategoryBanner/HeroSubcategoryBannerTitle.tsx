import { ReactNode } from 'react';
import { Heading } from '../../heading';

export type HeroSubcategoryBannerTitleProps = { children: ReactNode };

export const HeroSubcategoryBannerTitle = ({
	children,
}: HeroSubcategoryBannerTitleProps) => (
	<Heading fontSize="xxl" type="h1">
		{children}
	</Heading>
);
