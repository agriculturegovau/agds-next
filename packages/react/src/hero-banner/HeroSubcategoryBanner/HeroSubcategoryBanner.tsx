import { PropsWithChildren } from 'react';
import { HeroBannerBackground } from '../utils';
import { HeroSubcategoryBannerContainer } from './HeroSubcategoryBannerContainer';
import { HeroSubcategoryBannerContent } from './HeroSubcategoryBannerContent';

export type HeroSubcategoryBannerProps = PropsWithChildren<{
	/** The background of the component */
	background?: HeroBannerBackground;
}>;

export const HeroSubcategoryBanner = ({
	children,
	background = 'bodyAlt',
}: HeroSubcategoryBannerProps) => (
	<HeroSubcategoryBannerContainer background={background}>
		<HeroSubcategoryBannerContent>{children}</HeroSubcategoryBannerContent>
	</HeroSubcategoryBannerContainer>
);
