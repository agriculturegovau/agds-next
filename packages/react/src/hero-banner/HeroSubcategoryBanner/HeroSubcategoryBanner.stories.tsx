import { Meta, StoryObj } from '@storybook/react';
import { Breadcrumbs } from '../../breadcrumbs';
import { Box } from '../../box';
import {
	HeroSubcategoryBanner,
	HeroSubcategoryBannerProps,
} from './HeroSubcategoryBanner';
import { HeroSubcategoryBannerTitle } from './HeroSubcategoryBannerTitle';

type HeroSubcategoryBannerStoryProps = HeroSubcategoryBannerProps & {
	title: string;
	subtitle: string;
};

function Template({ title, ...props }: HeroSubcategoryBannerStoryProps) {
	return (
		<HeroSubcategoryBanner {...props}>
			<Breadcrumbs
				links={[
					{ href: '#', label: 'Section 1' },
					{ href: '#', label: 'Category page' },
					{ label: 'Subcategory page' },
				]}
			/>
			<HeroSubcategoryBannerTitle>{title}</HeroSubcategoryBannerTitle>
		</HeroSubcategoryBanner>
	);
}

const meta: Meta<HeroSubcategoryBannerStoryProps> = {
	title: 'layout/HeroBanner/HeroSubcategoryBanner',
	component: HeroSubcategoryBanner,
	parameters: {
		layout: 'fullscreen',
	},
	args: {
		title: 'Subcategory banner title - xxl/display (H1)',
	},
	render: Template,
};

export default meta;

type Story = StoryObj<HeroSubcategoryBannerStoryProps>;

export const Basic: Story = {
	args: {},
};

export const OnBodyAlt: Story = {
	name: 'On bodyAlt background',
	args: {
		background: 'body',
	},
	render: (args) => (
		<Box background="bodyAlt" paddingY={3}>
			<Template {...args} />
		</Box>
	),
};
