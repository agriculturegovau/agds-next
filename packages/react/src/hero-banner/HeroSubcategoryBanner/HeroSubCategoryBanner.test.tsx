import '@testing-library/jest-dom';
import 'html-validate/jest';
import { Breadcrumbs } from '../../breadcrumbs';
import { cleanup, render } from '../../../../../test-utils';
import { HeroSubcategoryBanner } from './HeroSubcategoryBanner';
import { HeroSubcategoryBannerTitle } from './HeroSubcategoryBannerTitle';

afterEach(cleanup);

function renderHeroSubcategoryBanner() {
	return render(
		<HeroSubcategoryBanner>
			<Breadcrumbs
				links={[
					{ href: '#', label: 'Section 1' },
					{ href: '#', label: 'Category page' },
					{ label: 'Subcategory page' },
				]}
			/>
			<HeroSubcategoryBannerTitle>
				Subcategory banner title - xxl/display (H1)
			</HeroSubcategoryBannerTitle>
		</HeroSubcategoryBanner>
	);
}

describe('HeroSubcategoryBanner', () => {
	it('renders correctly', () => {
		const { container } = renderHeroSubcategoryBanner();
		expect(container).toMatchSnapshot();
	});

	it('renders a valid HTML structure', () => {
		const { container } = renderHeroSubcategoryBanner();
		expect(container).toHTMLValidate({
			extends: ['html-validate:recommended'],
		});
	});
});
