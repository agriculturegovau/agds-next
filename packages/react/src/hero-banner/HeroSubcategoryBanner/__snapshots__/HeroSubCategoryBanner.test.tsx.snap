// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HeroSubcategoryBanner renders correctly 1`] = `
<div>
  <section
    class="css-1yg4yzk-boxStyles"
  >
    <div
      class="css-uq5mes-boxStyles"
    >
      <div
        class="css-1dasgjt-boxStyles"
      >
        <div
          class="css-uvibcs-boxStyles"
        >
          <nav
            aria-label="Breadcrumbs"
          >
            <ol
              class="css-1klpc3g-boxStyles-BreadcrumbsContainer"
            >
              <li
                class="css-1s21qr-boxStyles"
              >
                <svg
                  aria-hidden="true"
                  class="css-6n0j1b-BreadcrumbsDivider"
                  clip-rule="evenodd"
                  fill-rule="evenodd"
                  focusable="false"
                  height="24"
                  role="img"
                  viewBox="0 0 24 24"
                  width="24"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <polyline
                    points="9 18 15 12 9 6"
                  />
                </svg>
                <a
                  class="css-iuqbpq-TextLink"
                  href="#"
                >
                  Section 1
                </a>
              </li>
              <li
                class="css-1s21qr-boxStyles"
              >
                <svg
                  aria-hidden="true"
                  class="css-6n0j1b-BreadcrumbsDivider"
                  clip-rule="evenodd"
                  fill-rule="evenodd"
                  focusable="false"
                  height="24"
                  role="img"
                  viewBox="0 0 24 24"
                  width="24"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <polyline
                    points="9 18 15 12 9 6"
                  />
                </svg>
                <button
                  aria-expanded="false"
                  aria-label="Show all breadcrumbs"
                  class="css-1pa4jzb-BaseButton-boxStyles"
                  type="button"
                >
                  â€¦
                </button>
              </li>
              <li
                class="css-1s21qr-boxStyles"
              >
                <svg
                  aria-hidden="true"
                  class="css-6n0j1b-BreadcrumbsDivider"
                  clip-rule="evenodd"
                  fill-rule="evenodd"
                  focusable="false"
                  height="24"
                  role="img"
                  viewBox="0 0 24 24"
                  width="24"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <polyline
                    points="9 18 15 12 9 6"
                  />
                </svg>
                <a
                  class="css-iuqbpq-TextLink"
                  href="#"
                >
                  Category page
                </a>
              </li>
              <li
                class="css-1s21qr-boxStyles"
              >
                <svg
                  aria-hidden="true"
                  class="css-6n0j1b-BreadcrumbsDivider"
                  clip-rule="evenodd"
                  fill-rule="evenodd"
                  focusable="false"
                  height="24"
                  role="img"
                  viewBox="0 0 24 24"
                  width="24"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <polyline
                    points="9 18 15 12 9 6"
                  />
                </svg>
                Subcategory page
                <span
                  class="css-5ur8x"
                >
                   (current page)
                </span>
              </li>
            </ol>
          </nav>
          <h1
            class="css-1jdu126-boxStyles"
          >
            Subcategory banner title - xxl/display (H1)
          </h1>
        </div>
      </div>
    </div>
  </section>
</div>
`;
